internal/runner/runner.go

```diff
@@ -224,10 +224,11 @@ func (r *NomadJob) handleExitOrContextDone(ctx context.Context, cancelExecute co
 	// (tty has to be true) and converted to a SIGQUIT signal sent to the foreground process attached to the tty.
 	// By default, SIGQUIT causes the process to terminate and produces a core dump. Processes can catch this signal
 	// and ignore it, which is why we destroy the runner if the process does not terminate after a grace period.
-	n, err := stdin.Write([]byte{SIGQUIT})
-	if n != 1 {
-		log.WithField("runner", r.id).Warn("Could not send SIGQUIT because nothing was written")
-	}
+	_, err := stdin.Write([]byte{SIGQUIT})
+	// if n != 1 {
+	// This is always the case but the SIGQUIT gets received.
```

Comment:
Hm, this is an unexpected change for me.

Do you mean that the SIGQUIT is definitely sent even though the return value does not match our expectation of one byte? Why is that the case?

Reply:
Thank you for raising this still unclear point.
I agree that this is unusual, when normal behaviour would be more like [this](https://go.dev/play/p/GKSZCdKsOeX).

First of all, I only noticed that the now commented-out warning always appears during executions. Then I confirmed my suspicions with a test:
We add this test snippet in `ExecuteInteractively`:
``` Go
	go r.executeCommand(ctxExecute, command, stdin, stdout, stderr, exitInternal)
	go r.handleExitOrContextDone(ctx, cancelExecute, exitInternal, exit, stdin)

	go func() {
		<- time.After(10 * time.Second)
		n, err := stdin.Write([]byte{SIGQUIT})
		log.WithField("n", n).WithError(err).Info("We just send SIGQUIT")
	}()

	return exit, cancel, nil
```
We then execute sth like: `while true; do sleep 1; echo HI; done`
This results in the log statement:
```
time="2021-11-25T18:41:07.718215Z" level=info msg="We just send SIGQUIT" error="<nil>" n=0 package=runner
```
and the execution to be exited after the 10 seconds... But at this point I do not know more about the background.

Reply:
Indeed. I was able to confirm it and got the following output in CodeOcean:

```
while true; do sleep 1; echo HI; done

HI
HI
HI
HI
HI
HI
HI
HI
HI
^\
```

Obviously, the character is sent correctly and, as you mentioned, the stopping also works correctly. Regardless of the byte array, the length is always 0. 

Let's add (a link to) this discussion to the comment and accept it for now (if you don't have another idea).

```suggestion
	// The SIGQUIT is sent and correctly processed by the allocation.  However, for an unknown 
	// reason, the number of bytes written is always zero even though the error is nil. 
	// Hence, we disabled this sanity check here. See the MR for more details: 
	// https://github.com/openHPI/poseidon/pull/45#discussion_r757029024
```